src/app/components/content/admin/add-partner/add-partner.component.spec.ts

```diff
@@ -0,0 +1,153 @@
+        FormsModule,
+        ReactiveFormsModule
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA,
+      ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AddPartnerComponent);
+    component = fixture.componentInstance;
+    service = TestBed.inject(AuthenticateService);
+    element = fixture.debugElement.nativeElement;
+    mockHttpClient = TestBed.inject(HttpTestingController);
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should toggle admin value (True <=> False)', () => {
+    const onButton: HTMLButtonElement = element.querySelector('#onButton');
+    const offButton: HTMLButtonElement = element.querySelector('#offButton');
+
+    offButton.click();
+
+    expect(component.addMemberForm.controls.admin.value).toEqual(false);
+
+    onButton.click();
+
+    expect(component.addMemberForm.controls.admin.value).toEqual(true);
+  });
+
+  it('should check if email is valid', () => {
+    component.addMemberForm.controls.email.setValue('wrongEmail');
+    component.onSubmit();
+    expect(component.emailIsInvalid).toEqual(true);
+
+    component.addMemberForm.controls.email.setValue(TEST_USER.emailAddress);
+    component.onSubmit();
+    expect(component.emailIsInvalid).toEqual(false);
+  });
+
+  it('should check if password is valid', () => {
+    component.addMemberForm.controls.password.setValue('1234');
+    component.onSubmit();
+    expect(component.passwordIsInvalid).toEqual(true);
+
+    component.addMemberForm.controls.password.setValue(TEST_USER.password);
+    component.onSubmit();
+    expect(component.passwordIsInvalid).toEqual(false);
+  });
+
+  it('should check if confirm password is valid', () => {
+    component.addMemberForm.controls.password.setValue(TEST_USER.password);
+
+    component.addMemberForm.controls.confirmPassword.setValue('validPassword');
+    component.onSubmit();
+    expect(component.confirmPasswordIsInvalid).toEqual(true);
+
+    component.addMemberForm.controls.confirmPassword.setValue(TEST_USER.password);
+    component.onSubmit();
+    expect(component.confirmPasswordIsInvalid).toEqual(false);
+  });
+
+  it('should check if company name is valid', () => {
+    component.addMemberForm.controls.companyName.setValue('');
+    component.onSubmit();
+    expect(component.passwordIsInvalid).toEqual(true);
+
+    component.addMemberForm.controls.password.setValue(TEST_USER.name);
+    component.onSubmit();
+    expect(component.passwordIsInvalid).toEqual(false);
+  });
+
+  it('should check if form is valid', () => {
+    component.addMemberForm.controls.email.setValue(TEST_USER.emailAddress);
+    component.addMemberForm.controls.password.setValue(TEST_USER.password);
+    component.addMemberForm.controls.confirmPassword.setValue(TEST_USER.password);
+    component.addMemberForm.controls.companyName.setValue(TEST_USER.name);
+
+    component.onSubmit();
```

Comment:
when repeatedly do the same, you could put that in a function to prevent code duplications, increase readability etc.

Reply:
true was planned on doing it, kinda forgot XD